### PR TITLE
[card-22] fix: dart analysis info and warnings

### PR DESCRIPTION
**Pull request by minhajdevcrew**

### CODE REVIEW

**Comment by minhajdevcrew** [comment]
[Fix Dart Analysis issues](https://trello.com/c/g1ZMl9Zh/22-fix-dart-analysis-issues)

File: lib/text_parser.dart [hunk 1]
```diff
@@ -12,7 +12,7 @@ List<TextSegment> parseText(String? text) {
       unicode: true);
   final matches = exp.allMatches(text);
   var start = 0;
-  matches.forEach((match) {
+  for (var match in matches) {
```
Comment (arslandevcrew) [inline lib/text_parser.dart:15]: Please add some detail for words matches logic

File: lib/text_parser.dart [hunk 2]
```diff
@@ -25,27 +25,25 @@ List<TextSegment> parseText(String? text) {
...
       if (!isWord) {
-        return;
+        continue;
       }
 
       final isHashtag = keyword.startsWith('#');
       final isMention = keyword.startsWith('@');
       final isTag = keyword.startsWith('<');
-      var subString;
+      late String subString;
```
Comment (arslandevcrew) [inline lib/text_parser.dart]: Add some documentation for Regix logic

**Review by arslandevcrew — CHANGES_REQUESTED** [review]
Pease review the following changes

File: lib/text_parser.dart (hunk 1 above)
Comment (minhajdevcrew) [reply lib/text_parser.dart:15]: `// it means that there is some text before matched word e.g text = some text some text [matched word]
    if (match.start > start) {
      // add this text to [segments] e.g add "some text some text"
      if (segments.isNotEmpty && segments.last.isText) {
        segments.last.text += text.substring(start, match.start);
      } else {
        segments.add(TextSegment(text.substring(start, match.start)));
      }
      // set the [start] to the point where match was found 
      start = match.start;
    }`

File: lib/text_parser.dart (hunk 2 above)
Comment (minhajdevcrew) [reply lib/text_parser.dart]: `
// if there is ' ' or '\n' before keyword i.e is there is blank space or \n before @mention, #hashtag etc e.g " @mention", "\n @mention"
      final isWord = match.start == 0 ||
          [' ', '\n'].contains(text.substring(match.start - 1, start));
      // if there is not ' ' or '\n' then continue the loop e.g in the case "sometext@mention"
      if (!isWord) {
        continue;
      }
`

Comment (minhajdevcrew) [reply lib/text_parser.dart]: `
/// if keyword is custom tag e.g <tag>some text</tag>
      /// extract 'tag' and remove <tag>, </tag> from keyword using regex
      if (isTag) {
        final regexTag = RegExp(r"[<>]"); // <>
        final regexStartTag = RegExp(r"<\w+>"); // <tag>
        final regexEndTag = RegExp(r"<\/*\w*>*"); //</tag>
        tagName = keyword.replaceAll(regexTag, "");
        keywordToDisplay = keyword.replaceAll(regexStartTag, "");
        keywordToDisplay = keywordToDisplay.replaceAll(regexEndTag, "");
      } else {
        keywordToDisplay = keyword;
      }
`

**Review by arslandevcrew — APPROVED** [review]

**Pull request merged**